packages/protocol/contracts/layer2/based/anchor/ShastaAnchor.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "src/shared/based/LibSharedData.sol";
+import "../eip1559/LibEIP1559Classic.sol";
+import "./PacayaAnchor.sol";
+
+/// @title ShastaAnchor
+/// @notice Anchoring functions for the Shasta fork.
+/// @custom:security-contact security@taiko.xyz
+abstract contract ShastaAnchor is PacayaAnchor {
+    uint64 public immutable shastaForkHeight;
+
+    error InvalidForkHeight();
+
+    uint256[50] private __gap;
+
+    constructor(
+        address _resolver,
+        address _signalService,
+        uint64 _pacayaForkHeight,
+        uint64 _shastaForkHeight
+    )
+        PacayaAnchor(_resolver, _signalService, _pacayaForkHeight)
+    {
+        require(shastaForkHeight == 0 || shastaForkHeight > _pacayaForkHeight, InvalidForkHeight());
+        shastaForkHeight = _shastaForkHeight;
+    }
+
+    /// @notice Anchors the latest L1 block details to L2 for cross-layer
+    /// message verification.
+    /// @dev The gas limit for this transaction must be set to 1,000,000 gas.
+    /// @dev This function can be called freely as the golden touch private key is publicly known,
+    /// but the Taiko node guarantees the first transaction of each block is always this anchor
+    /// transaction, and any subsequent calls will revert with L2_PUBLIC_INPUT_HASH_MISMATCH.
+    /// @param _anchorBlockId The `anchorBlockId` value in this block's metadata.
+    /// @param _anchorStateRoot The state root for the L1 block with id equals `_anchorBlockId`.
+    /// @param _parentGasUsed The gas used in the parent block.
+    /// @param _baseFeeConfig The base fee configuration.
+    /// @param _signalSlots The signal slots to mark as received.
+    function shastaAnchor(
+        uint64 _anchorBlockId,
+        bytes32 _anchorStateRoot,
+        uint256 _parentBaseFee,
+        uint32 _parentGasUsed,
+        LibSharedData.BaseFeeConfig calldata _baseFeeConfig,
+        bytes32[] calldata _signalSlots
+    )
+        external
+        nonZeroBytes32(_anchorStateRoot)
+        nonZeroValue(_anchorBlockId)
+        nonZeroValue(_baseFeeConfig.gasIssuancePerSecond)
+        nonZeroValue(_baseFeeConfig.adjustmentQuotient)
+        onlyGoldenTouch
+        nonReentrant
+    {
+        require(shastaForkHeight != 0 && block.number >= shastaForkHeight, L2_FORK_ERROR());
```

Comment:
Shall we change it to `require(shastaForkHeight == 0 || block.number >= shastaForkHeight, L2_FORK_ERROR());` to allow Shasta fork from genesis? Since in `ShastaAnchor.constructor `, `shastaForkHeight == 0` is allowed.

Reply:
If shastaForkHeight == 0, it means shasta fork is not schedule to be active, it has to be 1 if we want to use shasta right after genesis.